components/formats-gpl/src/loci/formats/in/FlexReader.java

```diff
@@ -814,9 +813,13 @@ private void parseFlexFile(int currentWell, int wellRow, int wellCol,
       ifd = file.ifds.get(0);
     }
     else {
-      TiffParser parser = new TiffParser(file.file);
-      ifd = parser.getFirstIFD();
-      parser.getStream().close();
+        RandomAccessInputStream ras = new RandomAccessInputStream(file.file);
+        try {
+            TiffParser parser = new TiffParser(ras);
+            ifd = parser.getFirstIFD();
+        } finally {
+            if (ras != null) ras.close();
```

Comment:
* The null check is more a cut and
* The method  ```getFirstIFD``` can throw an exception => the stream might never be closed.
* intend: this is really a paint when switching between BF and omero. We probably need to sort that one out.